lib/cartopy/crs.py

```diff
@@ -2006,3 +2066,28 @@ def epsg(code):
     """
     import cartopy._epsg
     return cartopy._epsg._EPSGProjection(code)
+
+
+def from_proj4(proj4_terms, globe=None, bounds=None):
```

Comment:
Let's put the implementation in the _proj4 module and make this a stub function that calls the appropriate function there (``crs.py`` is already waaaay too long 😄).